packages/harpguru-core/src/components/option-stack/components/shift-in-stack/shift-in-stack.tsx

```diff
@@ -0,0 +1,126 @@
+import Animated, {
+  cond,
+  eq,
+  sub,
+  Node,
+  add,
+  interpolate,
+} from 'react-native-reanimated'
+import { TouchableOpacity } from 'react-native-gesture-handler'
+import React from 'react'
+import { AntDesign } from '@expo/vector-icons'
+
+import { getStyles } from '../../utils'
+import type {
+  OptionStackProps,
+  WithStateValue,
+  WithTransition,
+} from '../../types'
+import { useInterpolateTransitionValue } from '../../hooks'
+import { colors } from '../../../../styles'
+
+type PointerProperties = {
+  readonly prevInStack: () => void
+  readonly nextInStack: () => void
+  readonly prevPointerEvents: Node<'none' | 'auto'>
+  readonly nextPointerEvents: Node<'none' | 'auto'>
+  readonly prevPointerOpacity: Node<number>
+  readonly nextPointerOpacity: Node<number>
+}
+
+const usePointerProperties = (
```

Comment:
Consider pulling this out to the utils folder. It might feel a bit disjointed having it all the way back up there. Should still consider pulling these components back out to the top level so that they can legitimately have their own utils folder underneath them.

Reply:
I suspect that the next action will be to apply an option-stack approach to the covariance menu. That will be a great opportunity to remove all the other option components and only have these ones there. I could probably do it now but I think it will be much more convenient to do it after the other components are cleared out.